badge-maker/lib/badge-renderers.js

```diff
@@ -270,25 +178,172 @@ class Badge {
     this.label = label
     this.message = message
     this.accessibleText = accessibleText
-    this.renderedLogo = renderedLogo
-    this.renderedLabel = renderedLabel
-    this.renderedMessage = renderedMessage
+
+    this.logoElement = getLogoElement({
+      logo,
+      horizPadding,
+      badgeHeight: this.constructor.height,
+      logoWidth,
+    })
+    this.foregroundGroupElement = this.getForegroundGroupElement()
   }
 
   static render(params) {
     return new this(params).render()
   }
 
+  getTextElement({ leftMargin, content, link, color, textWidth, linkWidth }) {
```

Comment:
I think there's actually scope to factor out `getTextElement()` into a shared helper (maybe `getBackgroundGroupElement()` and `getForegroundGroupElement()` too), DRY up and have FTB share more logic with flat/flat-square/plastic. At the moment FTB and flat/flat-square/plastic use different attribute order in the XML output though so I think doing this will require tweaking the snapshots. For this reason I think this should be done in a follow-on PR instead of here. The property that we know the code exactly matches the snapshots before and after this PR is probably a valuable property for reviewing plus there is already enough stuff changing at once here.